src/Compilers/VisualBasic/Portable/Symbols/ReferenceManager.vb

```diff
@@ -856,15 +853,10 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
                 Protected Overrides Sub AddAvailableSymbols(assemblies As List(Of AssemblySymbol))
                     Dim internalsMayBeVisibleToCompilation = Me.InternalsMayBeVisibleToCompilation
 
-                    ' accessing cached symbols requires a lock
-                    SyncLock SymbolCacheAndReferenceManagerStateGuard
-                        For Each assemblySymbol In CachedSymbols
-                            Dim peAssembly = TryCast(assemblySymbol, PEAssemblySymbol)
-                            If IsMatchingAssembly(peAssembly) Then
-                                assemblies.Add(peAssembly)
-                            End If
-                        Next
-                    End SyncLock
+                    CachedSymbols.CopyTo(Of AssemblySymbol, AssemblyDataForFile)(
```

Comment:
>AssemblySymbol [](start = 44, length = 14)

```PEAssemblySymbol```?

Reply:
Can't be PEAssemblySymbol as the collection is List<AssemblySymbol>. The choice here is the perf of a cast or the complexity of yet another generic parameter.